Source/RakPeer.cpp

```diff
@@ -3595,7 +3747,39 @@ void RakPeer::SendBuffered( const char *data, int numberOfBitsToSend, PacketPrio
 #endif
 }
 // --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-bool RakPeer::SendImmediate( char *data, int numberOfBitsToSend, PacketPriority priority, PacketReliability reliability, char orderingChannel, PlayerID playerId, bool broadcast, bool useCallerDataAllocation, RakNetTimeNS currentTime )
+void RakPeer::SendBufferedToList( const char *data, int numberOfBitsToSend, PacketPriority priority, PacketReliability reliability, char orderingChannel, PlayerID* players, int playerCount )
+{
+	RakAssert(orderingChannel >=0 && orderingChannel < 32);
+
+	BufferedCommandStruct *bcs;
+
+#ifdef _RAKNET_THREADSAFE
+	rakPeerMutexes[bufferedCommands_Mutex].Lock();
+#endif
+
+	for (int index = 0; index < playerCount; index++)
+	{
+		char* dataToBeWritten = new char[BITS_TO_BYTES(numberOfBitsToSend)];
+		RakAssert(dataToBeWritten);
+		memcpy(dataToBeWritten, data, BITS_TO_BYTES(numberOfBitsToSend));
+		bcs = bufferedCommands.WriteLock();
+		bcs->data = dataToBeWritten; // Making a copy doesn't lose efficiency because I tell the reliability layer to use this allocation for its own copy
+		bcs->numberOfBitsToSend = numberOfBitsToSend;
+		bcs->priority = priority;
+		bcs->reliability = reliability;
+		bcs->orderingChannel = orderingChannel;
+		bcs->listData.playerIds = players;
+		bcs->listData.playerCount = playerCount;
+		bcs->command = BufferedCommandStruct::BCS_SEND_LIST;
+		bufferedCommands.WriteUnlock();
+	}
+	
+#ifdef _RAKNET_THREADSAFE
+	rakPeerMutexes[bufferedCommands_Mutex].Unlock();
+#endif
+}
+// --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+bool RakPeer::SendImmediate(char* data, int numberOfBitsToSend, PacketPriority priority, PacketReliability reliability, char orderingChannel, PlayerID playerId, bool broadcast, bool useCallerDataAllocation, RakNetTimeNS currentTime, PlayerID* playerIds, int playerCount)
```

Comment:
Rather than having separate `playerId` and `playerIds` parameters, would it be better to call this as:

```c
SendImmediate((char*)bcs->data, bcs->numberOfBitsToSend, bcs->priority, bcs->reliability, bcs->orderingChannel, bcs->singleData.broadcast, true, timeNS, &bcs->singleData.playerId, 1);
```

I.e. pass a "list" with only a single element for sending to a single player.  It would reduce the complexity and duplication within this function at least.